manifests/server/export.pp

```diff
@@ -12,116 +14,89 @@
 #   NFSv3 clients.
 #
-# @param refer [String] A list of alternative locations for the filesystem.
+# @param refer A list of alternative locations for the filesystem.
 #
-# @param sec [Array] An Array of security flavors, in order of preference.
+# @param sec An Array of security flavors, in order of preference.
 #
-# @param no_root_squash [Boolean] If set, disable root squashing. This should
+# @param no_root_squash If set, disable root squashing. This should
 #   only be done if you *really* know what you are doing!
 #
-# @param all_squash [Boolean] If set, map all uids and gids to the `anonymous`
+# @param all_squash If set, map all uids and gids to the `anonymous`
 #   user.
 #
-# @param anonuid [String] Explicity set the UID of the `anonymous` user.
+# @param anonuid Explicity set the UID of the `anonymous` user.
 #
-# @param anongid [String] Explicity set the GID of the `anonymous` user.
+# @param anongid Explicity set the GID of the `anonymous` user.
 
-# @param custom [String] A custom set of options.  If set, all other options
+# @param custom A custom set of options.  If set, all other options
 #   will be ignored. `$mountpoint` and `$client` must still be set.  Do *not*
 #   include the parenthesis if you are writing a custom options string.
 #
 # @author Trevor Vaughan <mailto:tvaughan@onyxpoint.com>
 #
 define nfs::server::export (
-  $export_path,
-  $client,
-  $comment = '',
-  $insecure = false,
-  $rw = false,
-  $async = false,
-  $no_wdelay = false,
-  $nohide = false,
-  $crossmnt = false,
-  $subtree_check = false,
-  $insecure_locks = false,
-  $mountpoint = '',
-  $fsid = '',
-  $nordirplus = false,
-  $refer = '',
-  $sec = ['sys'],
-  $no_root_squash = false,
-  $all_squash = false,
-  $anonuid = '65534',
-  $anongid = '65534',
-  $custom = ''
+  Stdlib::Absolutepath                            $export_path,
+  Array[String]                                   $client,
+  Optional[String]                                $comment        = undef,
+  Boolean                                         $insecure       = false,
+  Boolean                                         $rw             = false,
+  Boolean                                         $async          = false,
+  Boolean                                         $no_wdelay      = false,
+  Boolean                                         $nohide         = false,
+  Boolean                                         $crossmnt       = false,
+  Boolean                                         $subtree_check  = false,
+  Boolean                                         $insecure_locks = false,
+  Optional[Variant[Boolean,Stdlib::Absolutepath]] $mountpoint     = undef,
+  Optional[String]                                $fsid           = undef,
+  Boolean                                         $nordirplus     = false,
+  Optional[String]                                $refer          = undef,
+  Tuple[Enum['sys','krb5','krb5i','krb5p'],1,2]   $sec            = ['sys'],
```

Comment:
The idea was min and max, so both. However, this should actually check to make sure only one type of krb* is passed.  Sys and one type of krb5 is fine, but not >1 krb5.